File: 0463-island-perimeter/0463-island-perimeter.java
class Solution {
    public int islandPerimeter(int[][] g) {
        
        boolean b[][]=new boolean[g.length][g[0].length];
        for(int i=0;i<g.length;i++){
            for (int j=0;j<g[0].length;j++){
                if(g[i][j]==1 && !b[i][j]){
                    return dfs(i,j,g,b);
                }
            }
        }
        return 0;
    }
    public int dfs(int i,int j,int[][] g,boolean[][] b){
        if(i<0 || j<0 || i>=g.length || j>=g[0].length){
            return 1;
        }
        if(g[i][j]==0) return 1;
        if(b[i][j]) return 0;
        
        b[i][j]=true;
        int ans=0;
        int[] d1={0,1,0,-1};
        int[] d2={1,0,-1,0};
        for(int x=0;x<4;x++){
            int r=i+d1[x];
            int c=j+d2[x];
            ans+=dfs(r,c,g,b);
        }
        return ans;
    }
}

File: 0463-island-perimeter/0463-island-perimeter.py
class Solution:
    def islandPerimeter(self, g: List[List[int]]) -> int:
        
        def tra(i,j):
            if i<0 or j<0 or i>len(g)-1 or j>len(g[0])-1 or g[i][j]==0:
                return 1
            elif self.b[i][j]==False:
                return 0
            a=0
            self.b[i][j]=False
            a+=tra(i-1,j)
            a+=tra(i,j-1)
            a+=tra(i+1,j)
            a+=tra(i,j+1)
            return a



        self.b=[[[False] for _ in range(len(g[0]))]for _ in range(len(g))]
        a,b=0,0
        for i in range(len(g)):
            for j in range(len(g[i])):
                if g[i][j]==1:
                    a,b=i,j
                    self.b[i][j]=True
        return tra(a,b)